### PR TITLE
Symfony 5.1 compatibility

### LINKED ISSUE #7: Symfony 5.1 compatibility
In Symfony 5.1, the security and especially the Guard system has been revamped.

Read: 
* https://symfony.com/blog/new-in-symfony-5-1-updated-security-system
* http://wouterj.nl/2020/04/authenticators-new-symfony-security

This issue is to keep track of the evolution and compatibility of this package with Symfony 5.1.

I had a conversation with @wouterj on Slack and this is the outcome:
* Deprecated the custom Guard provided by this package, deprecated in favor of the new system.
* Both mechanism will be available in until the Guards are removed from Symfony (removed in Symfony 6)

### PR DESCRIPTION
**Pull request by drupol**
This PR adds compatibility with Symfony > 5.1

* [x] Add a new `CasAuthenticator` implementing the new interface `Symfony\Component\Security\Http\Authenticator\AuthenticatorInterface`
* [x] Update the documentation with new configuration
* [ ] Add tests

TODO:
* [ ] Revert the changes in composer.json(`minimum-stability, prefer-stable`) when Symfony 5.1 is released

Fixes #7

### CODE REVIEW

File: src/Security/CasAuthenticator.php [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            throw new AuthenticationException($exception->getMessage());
+        }
+
+        if (false === ($introspect instanceof ServiceValidate)) {
+            throw new AuthenticationException(
+                'Failure in the returned response'
+            );
+        }
+
+        return new Passport(new CasUser($introspect->getCredentials()), new CasUserCredentials());
```
[the remark this answers is not in the document]
Comment (drupol) [reply src/Security/CasAuthenticator.php]: Thanks :-)

**Comment by IonBazan** [comment]
@drupol I believe this can be closed as the changes are already on `master`

**Comment by drupol** [comment]
No, not yet, I will try to dedicate some time next week to update this branch.

**Comment by drupol** [comment]
Superseded by #59

**Pull request closed without merging**